src/WorkflowCore/Interface/IWorkflowMiddleware.cs

```diff
@@ -0,0 +1,41 @@
+using System.Threading.Tasks;
+using WorkflowCore.Models;
+
+namespace WorkflowCore.Interface
+{
+    /// <summary>
+    /// Determines at which point to run the middleware.
+    /// </summary>
+    public enum WorkflowMiddlewarePhase
```

Comment:
I had considered also having a `WorkflowMiddlewarePhase.Both` to enable middleware that runs both pre/post. Do you think that would be a good idea?